pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -74,53 +74,64 @@ func (s *Scheduler) Solve(ctx context.Context, provisioner *v1alpha5.Provisioner
 	if err := s.Topology.Inject(ctx, constraints, pods); err != nil {
 		return nil, fmt.Errorf("injecting topology, %w", err)
 	}
-	// Separate pods into schedules of isomorphic scheduling constraints.
-	schedules, err := s.getSchedules(ctx, constraints, pods)
+	instanceTypes, err := s.CloudProvider.GetInstanceTypes(ctx, constraints.Provider)
 	if err != nil {
-		return nil, fmt.Errorf("getting schedules, %w", err)
+		return nil, fmt.Errorf("getting instance types, %w", err)
 	}
-	return schedules, nil
+	// Separate pods into schedules of isomorphic scheduling constraints.
+	return s.getSchedules(constraints, instanceTypes, pods), nil
 }
 
 // getSchedules separates pods into a set of schedules. All pods in each group
 // contain isomorphic scheduling constraints and can be deployed together on the
 // same node, or multiple similar nodes if the pods exceed one node's capacity.
-func (s *Scheduler) getSchedules(ctx context.Context, constraints *v1alpha5.Constraints, pods []*v1.Pod) ([]*Schedule, error) {
-	// schedule uniqueness is tracked by hash(Constraints)
-	schedules := map[uint64]*Schedule{}
+func (s *Scheduler) getSchedules(constraints *v1alpha5.Constraints, instanceTypes []cloudprovider.InstanceType, pods []*v1.Pod) []*Schedule {
+	schedules := []*Schedule{}
 	for _, pod := range pods {
-		if err := constraints.ValidatePod(pod); err != nil {
-			logging.FromContext(ctx).Infof("Unable to schedule pod %s/%s, %s", pod.Namespace, pod.Name, err)
-			continue
+		isCompatible := false
+		for _, schedule := range schedules {
+			if err := schedule.Requirements.Compatible(v1alpha5.NewPodRequirements(pod)); err == nil {
+				// Test if there is any instance type that can support the combined constraints
+				// TODO: Implement a virtual node approach solution that combine scheduling and node selection to solve this problem.
+				c := schedule.Tighten(pod)
```

Comment:
This test would fail currently since the instance type checking isn't done.  Previously they would get different schedules due to hashing, so it wasn't an issue. 

```go
It("should launch instances for Nvidia and AMD GPU resources on different instances", func() {
				nodeNames := sets.NewString()
				for _, pod := range ExpectProvisioned(ctx, env.Client, selectionController, provisioners, provisioner,
					test.UnschedulablePod(test.PodOptions{
						ResourceRequirements: v1.ResourceRequirements{
							Requests: v1.ResourceList{resources.NvidiaGPU: resource.MustParse("1")},
							Limits:   v1.ResourceList{resources.NvidiaGPU: resource.MustParse("1")},
						},
					}),
					// Should pack onto a different instance
					test.UnschedulablePod(test.PodOptions{
						ResourceRequirements: v1.ResourceRequirements{
							Requests: v1.ResourceList{resources.AMDGPU: resource.MustParse("1")},
							Limits:   v1.ResourceList{resources.AMDGPU: resource.MustParse("1")},
						},
					})) {
					node := ExpectScheduled(ctx, env.Client, pod)
					nodeNames.Insert(node.Name)
				}
				Expect(nodeNames.Len()).To(Equal(2))
			})
```